### DISCUSSION TITLE
Can the CF Workers + D1 server be made open source?

### DISCUSSION THREAD

**Discussion by tylermercer** [evoluhq/evolu#335] [Q&A] [answered]
The docs mention "a free `evolu.world` server built on the Cloudflare Workers and D1 database for testing and personal usage." Any chance that could be open-sourced so it can be self-hosted? I checked the evolu-server directory but it looks like the app there is Express + SQLite.

Evolu looks awesome, by the way. Excited to build something on it.

**Comment by steida** [comment, marked as answer]
It will be open-sourced when D1 will be GA. We need genuine transactions, not just batched. Now it's hacked.

**Reply by danilaplee** [reply]
this has already happened today

**Reply by steida** [reply]
No, there are still no genuine transactions yet. You can use https://github.com/evoluhq/evolu/tree/main/packages/evolu-server

**Reply by danilaplee** [reply]
I dont want to run any servers, I killed my database server after moving to evolu from couchbase.
ok, I can wait until you release the wrangler worker :)

**Reply by acoreyj** [reply]
> It will be open-sourced when D1 will be GA. We need genuine transactions, not just batched. Now it's hacked.

What's the problem with releasing the hacked cloudflare server? How does it differ?

Also they say batched statements are transactions and they previewed a feature a couple years ago but haven't said much so not sure it will change beyond that due to the architecture of D1. 

https://developers.cloudflare.com/d1/build-with-d1/d1-client-api/

https://blog.cloudflare.com/whats-new-with-d1

**Reply by steida** [reply]
@acoreyj It's a hack because the Merkle Tree is rebuilt on every insert, which is super inefficient. I'm working hard on the new, much faster, and scalable sync that will work even with Cloudflare D1 as is. ETA 2 weeks I hope.

**Reply by abhishekvash** [reply]
@steida Any updates on this?

**Reply by steida** [reply]
Sooooon. I'm close to release. Sorry for the delay 😇

**Reply by njg4ne** [reply]
Also wondering about this!

**Reply by steida** [reply]
Fixed to work with Cloudflare a long time age. Release probably today.

**Comment by danilaplee** [comment]
hey, can u make this open source because it seems i used up your quota, I'am very sorry

**Reply by steida** [reply]
There is a lot of quota :-)
